Failures in tests with marker `obolibrary`

Recent test [failures](https://github.com/dandi/dandi-cli/actions/runs/20180629885/job/58024975882?pr=1760) on tests with marker `obolibrary` seems to be cause by content at address such as http://purl.obolibrary.org/obo/NCBITaxon_1234175 is not XML but HTML, and the `xml.dom.minidom.parseString` function is not able to parse such content.

```console
/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/site-packages/tenacity/__init__.py:480: in __call__
    result = fn(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^
dandi/metadata/util.py:430: in parse_purlobourl
    doc = parseString(req.text)
          ^^^^^^^^^^^^^^^^^^^^^
/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/xml/dom/minidom.py:2009: in parseString
    return expatbuilder.parseString(string)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/xml/dom/expatbuilder.py:922: in parseString
    return builder.parseString(string)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/xml/dom/expatbuilder.py:220: in parseString
    parser.Parse(string, True)
E   xml.parsers.expat.ExpatError: syntax error: line 1, column 0
```

However, it seems that the content at http://purl.obolibrary.org/obo/NCBITaxon_1234175 has always been HTML. How was the `xml.dom.minidom.parseString` function able to parse the HTML content at the page previously? Should we instead change to use addresses such as https://ontobee.org/ontology/rdf/NCBITaxon?iri=http://purl.obolibrary.org/obo/NCBITaxon_1234175, which contains comparable RDF document?

--- a/dandi/metadata/util.py
+++ b/dandi/metadata/util.py
@@ -425,7 +425,11 @@ def parse_purlobourl(
     :return: dictionary containing found nodes
     """
 
-    req = requests.get(url, allow_redirects=True)
+    if url.startswith(NCBITAXON_URI_TEMPLATE.format("")):
+        rdf_content_url = f"https://ontobee.org/ontology/rdf/NCBITaxon?iri={url}"
+        req = requests.get(rdf_content_url, allow_redirects=True)
+    else:
+        req = requests.get(url, allow_redirects=True)
     req.raise_for_status()
     doc = parseString(req.text)
     for elfound in doc.getElementsByTagName("Class"):

